File: Conceitos de LPs/Tipos de Dados/Conjunto/Conjunto.cs
//CSharp

using System;
using System.Collections.Generic;

class Conjunto {
    static void Main() {
        // Definindo e inicializando conjuntos
        HashSet<int> conjuntoA = new HashSet<int> { 1, 2, 3, 4, 5 };
        HashSet<int> conjuntoB = new HashSet<int> { 4, 5, 6, 7, 8 };

        // União
        HashSet<int> uniao = new HashSet<int>(conjuntoA);
        uniao.UnionWith(conjuntoB);

        // Intersecção
        HashSet<int> intersecao = new HashSet<int>(conjuntoA);
        intersecao.IntersectWith(conjuntoB);

        // Diferença
        HashSet<int> diferenca = new HashSet<int>(conjuntoA);
        diferenca.ExceptWith(conjuntoB);

        // Exibindo resultados
        Console.WriteLine("União: " + string.Join(", ", uniao));
        Console.WriteLine("Intersecção: " + string.Join(", ", intersecao));
        Console.WriteLine("Diferença: " + string.Join(", ", diferenca));
    }
}

File: Conceitos de LPs/Tipos de Dados/Conjunto/conjunto.cpp
//C++
#include <iostream>
#include <set>

using namespace std;

int main() {
    // Definindo e inicializando conjuntos
    set<int> conjuntoA = {1, 2, 3, 4, 5};
    set<int> conjuntoB = {4, 5, 6, 7, 8};

    // União
    set<int> uniao;
    set_union(conjuntoA.begin(), conjuntoA.end(),
                   conjuntoB.begin(), conjuntoB.end(),
                   inserter(uniao, uniao.begin()));

    // Intersecção
    set<int> intersecao;
    set_intersection(conjuntoA.begin(), conjuntoA.end(),
                          conjuntoB.begin(), conjuntoB.end(),
                          inserter(intersecao, intersecao.begin()));

    // Diferença
    set<int> diferenca;
    set_difference(conjuntoA.begin(), conjuntoA.end(),
                        conjuntoB.begin(), conjuntoB.end(),
                        inserter(diferenca, diferenca.begin()));

    // Exibindo resultados
    cout << "União: ";
    for (int i : uniao) cout << i << " ";
    cout << "\nIntersecção: ";
    for (int i : intersecao) cout << i << " ";
    cout << "\nDiferença: ";
    for (int i : diferenca) cout << i << " ";
    cout << endl;

    return 0;
}
